playwright/tests/desk-tool/documentActions.spec.ts

```diff
@@ -0,0 +1,113 @@
+import {test, expect} from '@playwright/test'
+import {createUniqueDocument, DEFAULT_DATASET, deleteDocument} from '../../helpers'
+
+/* 
+  **NOTE**
+  In these tests, we are not testing input events, so to speed up the tests we will use use `page.fill()` instead of `page.type()` to populate the document title field 
+*/
+
+const DOC_TITLE = 'Document actions [Playwright]'
+const inputSelector = 'data-testid=input-title >> input'
+
+// The path of the network request we want to wait for
+const documentActionPath = (action) =>
+  `*/**/mutate/${DEFAULT_DATASET}?tag=sanity.studio.document.${action}*`
+
+// Helper to generate path to document created in a test
+const generateDocumentUri = (docId) =>
+  `/${DEFAULT_DATASET}/desk/input-ci;documentActionsCI;${docId}`
+
+// Helper to get the path to the document the client creates for a test
+async function getDocumentUri(_type: string, title?: string) {
+  const testDoc = await createUniqueDocument({_type, title})
+  return [generateDocumentUri(testDoc?._id), testDoc?._id]
+}
+
+test.describe('@sanity/desk-tool: document actions', () => {
+  // Before we run our tests, delete all previously created test documents to avoid clutter
+  test.beforeAll(async () => {
+    await deleteDocument({query: `*[_type == "documentActionsCI"]`})
+  })
+
+  test('edit and publish document', async ({page}) => {
+    const publishButton = page.locator('data-testid=action-Publish')
+
+    await getDocumentUri('documentActionsCI').then(([path]) => {
+      page.goto(path)
+    })
```

Comment:
With my above suggestion, this would be something like:

```suggestion
    const {studioPath} = await createDocument('documentActionsCI')
    await page.goto(path)
```

Currently it does not return the `goto`, which I assume is a promise